bootstrap/src/intnb.ml

```diff
@@ -1,12 +1,72 @@
+open Rudiments_functions
 open Intnb_intf
 
+module Make_derived (T : I_derived) : S_derived with type t := T.t = struct
```

Comment:
I debated whether to extract this code into a functor, but after repeatedly fixing issues in four nearly identical copies (and sometimes missing one or more copies) I decided to go for it.  The `**` operator could also be extracted into this functor, but doing so would require the functor to additionally know whether it is used in a signed context.  Thoughts on that one?